### PR TITLE
Fix bug where actions are not passed to components

### LINKED ISSUE #2458: Card actions are not firing
**What steps did you take and what happened:**
Cards have to ability to embed forms and use a component action. These currently broke with the form changes.

**What did you expect to happen:**
Maybe we shouldn't use this at all since such patterns have not been seen outside of Octant internals, and it is not discoverable to users.

**Environment:**

- Octant version (use `octant version`): 0.20

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
**What this PR does / why we need it**:
Non-numbered action form fields were filtered by the transform function.

**Which issue(s) this PR fixes**
- Fixes #2458 

Signed-off-by: Sam Foo <foos@vmware.com>

### CODE REVIEW

File: web/src/app/modules/shared/models/form-helper.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { FormHelper } from './form-helper';
+
+describe('FormHelper', () => {
```
Comment (xtreme-vikram-yadav) [inline web/src/app/modules/shared/models/form-helper.spec.ts:3]: Since we are testing return value, we can easily add tests for values with `empty string`, `stringified number` and `NaN`.

Comment (GuessWhoSamFoo) [reply web/src/app/modules/shared/models/form-helper.spec.ts:3]: done

**Review by wwitzel3 — APPROVED** [review]
